packages/integration-tests/tests/social-session.test.ts

```diff
@@ -0,0 +1,121 @@
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { mockSocialConnectorId, mockSocialConnectorConfig } from '@/__mocks__/connectors-mock';
+import {
+  signInWithSocial,
+  getAuthWithSocial,
+  registerWithSocial,
+  bindWithSocial,
+  signInWithUsernameAndPassword,
+} from '@/api';
+import MockClient from '@/client';
+import { setUpConnector, createUserByAdmin } from '@/helpers';
+import { generateUsername, generatePassword } from '@/utils';
+
+const state = 'foo_state';
+const redirectUri = 'http://foo.dev/callback';
+const code = 'auth_code_foo';
+
+describe('social sign-in and register', () => {
+  const userId = crypto.randomUUID();
+
+  beforeAll(async () => {
+    await setUpConnector(mockSocialConnectorId, mockSocialConnectorConfig);
+  });
+
+  it('register with social', async () => {
+    const client = new MockClient();
+
+    await client.initSession();
+    assert(client.interactionCookie, new Error('Session not found'));
+
+    await expect(
+      signInWithSocial(
+        { state, connectorId: mockSocialConnectorId, redirectUri },
+        client.interactionCookie
+      )
+    ).resolves.toBeTruthy();
+
+    const response = await getAuthWithSocial(
+      { connectorId: mockSocialConnectorId, data: { state, redirectUri, code, userId } },
+      client.interactionCookie
+    ).catch((error: unknown) => error);
+
+    // User with social does not exist
+    expect(response instanceof HTTPError && response.response.statusCode === 422).toBe(true);
+
+    // Register with social
+    const { redirectTo } = await registerWithSocial(
+      mockSocialConnectorId,
+      client.interactionCookie
+    );
+
+    await client.processSession(redirectTo);
+
+    expect(client.isAuthenticated).toBeTruthy();
+  });
+
+  /* Note: As currently we can not prepare a social identities through admin api.
+   * The sign-in test case MUST run concurrently after the register test case
+   */
```

Comment:
Follow a better comment style?
```suggestion
  /*
   * Note: As currently we can not prepare a social identities through admin api.
   * The sign-in test case MUST run concurrently after the register test case
   */
```
See https://google.github.io/styleguide/jsguide.html#formatting-block-comment-style

Reply:
updated